core/src/test/scala/unit/kafka/integration/BaseTopicMetadataTest.scala

```diff
@@ -253,17 +253,19 @@ abstract class BaseTopicMetadataTest extends ZooKeeperTestHarness {
 
     // Assert that topic metadata at new brokers is updated correctly
     servers.filter(x => x.brokerState.currentState != NotRunning.state).foreach(x =>
-      waitUntilTrue(() =>
-        topicMetadata == ClientUtils.fetchTopicMetadata(
-          Set.empty,
-          Seq(new Broker(x.config.brokerId,
-            x.config.hostName,
-            x.boundPort()).getBrokerEndPoint(SecurityProtocol.PLAINTEXT)),
-          "TopicMetadataTest-testBasicTopicMetadata",
-          2000, 0), "Topic metadata is not correctly updated"))
+      waitUntilTrue(() => {
+          val foundMetadata = ClientUtils.fetchTopicMetadata(
+            Set.empty,
+            Seq(new Broker(x.config.brokerId,
+              x.config.hostName,
+              x.boundPort()).getBrokerEndPoint(SecurityProtocol.PLAINTEXT)),
+            "TopicMetadataTest-testBasicTopicMetadata", 2000, 0)
+          topicMetadata.brokers.sortBy(_.id) == foundMetadata.brokers.sortBy(_.id) &&
+            topicMetadata.topicsMetadata.sortBy(_.topic) == foundMetadata.topicsMetadata.sortBy(_.topic)
```

Comment:
@ijuma I haven't seen an issue with test failure because of it. I did see issues with brokers and topics. I had considered changing the `equals` implementation, but since this is legacy just wanted minimal change until completely migrated. Perhaps the `equals` implementations in the o.a.k.c classes should be evaluated though.